tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -148,8 +150,19 @@ def publish_functions_package(cls, function_params, package):
 
             # upload package
             blob_name = '%s.zip' % function_params.function_app_name
-            blob_client.create_blob_from_path(
-                FUNCTION_CONSUMPTION_BLOB_CONTAINER, blob_name, package.pkg.path)
+
+            # Windows requires temporary flag to access file
+            # and to open the file as stream vs using the path
+            if os.name == 'nt':
```

Comment:
Can we lift this "opening package" code into the Packaging class, so it can be reused in the other providers too?

Reply:
Hi Andy, I agree we should make it so it can be re-used by other providers. Would it make more sense to have this in the utils vs function package?

Reply:
@aluong code has been updated per your suggestion and all tests are passing. Could you review and provide any feedback?

Reply:
This looks good. thank you.